### PR TITLE
Remove deprecated popup and dialog components

### LINKED ISSUE #852: Dialogs are shown even from New...Dialog calls without .Show()
### Describe the bug:
Dialogs are all shown immediately despite the fact that New...Dialog is not intended to show.

### To Reproduce:
Steps to reproduce the behaviour:
1. Call `dialog.NewConfirm(...)`
2. See that it is visible immediately

### Example code:
```go
dialog.NewConfirm("Confirmation", "Are you enjoying this demo?", confirmCallback, win)
```

### Device (please complete the following information):
 - OS: Mac, Linux
 - Version: latest
 - Go version: 1.12, 1.13
 - Fyne version: 1.2.x, develop

This is due to the underlying `NewModalPopUp` showing immediately.
Therefore we should not create the popup until Show() is called. This makes sense anyhow.
There is too much display code in the setup methods.
The attached patch would fix the issue but is ugly, I think that the whole `dialog` package needs work so that it's more semantic based.

### PR DESCRIPTION
**Pull request by andydotxyz**
Fix it so that dialogs now don't show until Show or ShowXxx is called
Fixes #852

### Checklist:

- [x] Tests included. <- existing tests, this is removing old code
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
- [x] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

**Review by Jacalz — COMMENTED** [review]
Looks to be mostly good to me. Was it intended to add the `cmd/fyne_demo/Icon.png` file?
I guess the changes here should probably also be added to the changelog (including the removal of dialog.FileIcon)?

**Comment by andydotxyz** [comment]
Thanks for those hints @Jacalz - the icon was not supposed to be in.
Resolved and CHANGELOG updated.

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,9 @@ These changes likely break some apps, please read the
 * A new, more flexible, Theme interface has been created that we encourage developers to use
 * The desktop.Cursor definition was renamed desktop.StandardCursor to make way for custom cursors
 
+* Dialogs no longer show when created, unless using the ShowXxx convenience methods
+* Removed deprecated types including dialog.FileIcon
```
Comment (Jacalz) [inline CHANGELOG.md:19]: You probably should mention that they should use widget.FileIcon instead?

**Review by Jacalz — CHANGES_REQUESTED** [review]
Sorry, just one little comment on the changelog.

File: CHANGELOG.md (hunk 1 above)
Comment (andydotxyz) [reply CHANGELOG.md]: Fixed

**Review by Jacalz — APPROVED** [review]

**Review by toaster — APPROVED** [review]
LGTM

**Pull request merged**